src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -147,7 +147,7 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
     }
 
     private async _executeWithVaulted(payment: OrderPaymentRequestBody): Promise<InternalCheckoutSelectors> {
-        if (this._isHostedPaymentFormEnabled(payment.methodId)) {
+        if (this._isHostedPaymentFormEnabled(payment.methodId) && this._hostedForm) {
```

Comment:
Is there a scenario where hostedForm will not be available?

Reply:
@bc-akshatkaushik  What is validated before this step is if the payment is done with an instrument or with credit card, the hosted form Its not validated before.

Reply:
@bc-akshatkaushik

Reply:
In this line https://github.com/bigcommerce/checkout-sdk-js/pull/1223/files#diff-d755dc9987b2df0ced566e63258108d7af6892bc592ed1cccd9f4ddbe119dfc9R167 we check for `this._isHostedPaymentFormEnabled(methodId)` which is what you are checking above as well. Hence checking why the double check?
```suggestion
        if (this._hostedForm) {
```